app/jobs/generate_uniprot_mapping_tsv.rb

```diff
@@ -0,0 +1,54 @@
+class GenerateUniprotMappingTsv < GenerateTsvs
+    def perform
+      ensure_downloads_directory_exists
+      tsvs_to_generate.each do |e|
+        begin
+          tmp_file = tmp_file(e.file_name)
+          tmp_file.puts(e.headers.join("\t"))
+  
+          e.objects.find_each do |object|
```

Comment:
You probably don't need the indirection here of calling `e.objects` and can delete `def self.objects` in the presenter file. This can probably just be `Gene.find_each`